### PR TITLE
feat: add validation to scopesets and allow unknown ones

### PR DESCRIPTION
**Pull request by alexche-cchange**
asana: [Add all impact definitions from OpenEPD spreadsheet to spec](https://app.asana.com/0/1203527467850668/1208021404558576/f)

### CODE REVIEW

File: src/openepd/model/lcia.py [hunk 1]
```diff
@@ -194,17 +197,43 @@ class ScopeSet(BaseOpenEpdSchema):
         description="Potential net benefits from reuse, recycling, and/or energy recovery beyond the system boundary.",
     )
 
+    @pyd.root_validator
+    def _all_measurements_should_be_same_unit_validator(cls, values):
```
Comment (alexlitvinenko) [inline src/openepd/model/lcia.py:201]: add type hints

File: src/openepd/model/lcia.py [hunk 2]
```diff
@@ -213,39 +242,121 @@ def get_scopeset_by_name(self, name: str) -> ScopeSet | None:
...
         for f_name, f in self.__fields__.items():
             if f.alias == name:
                 return getattr(self, f_name)
             if f_name == name:
                 return getattr(self, f_name)
+        # probably unknown impact, coming from 'extra' fields
+        return getattr(self, name, None)
+
+    @pyd.root_validator(skip_on_failure=True)
+    def _extra_scopeset_validator(cls, values):
```
Comment (alexlitvinenko) [inline src/openepd/model/lcia.py:255]: add type hints

**Review by alexlitvinenko — COMMENTED** [review]
You've changed type for field, e.g. `ScopeSet` -> `ScopeSetGwp`. This is major change.

**Review by corvis — APPROVED** [review]

**Comment by alexche-cchange** [comment]
Discussed with Oleg: 

a) need a proper validation via pint and late building
b) need a support for two distinct units (like AP and AP_H)
c) need solution from Phil about m3AWARE

**Comment by alexche-cchange** [comment]
> Discussed with Oleg:
> 
> a) need a proper validation via pint and late building b) need a support for two distinct units (like AP and AP_H) c) need solution from Phil about m3AWARE

Done all three.

**Pull request merged**